client-misk-hibernate/src/main/kotlin/app/cash/backfila/client/misk/hibernate/internal/BoundingRangeStrategy.kt

```diff
@@ -104,29 +130,52 @@ private fun <E : DbEntity<E>, Pkey : Any> selectMaxBound(
   backfill: HibernateBackfill<E, Pkey, *>,
   session: Session,
   schemaAndTable: String,
-  previousEndKey: ByteString?,
-  backfillRange: KeyRange,
-  scanSize: Long?
+  previousEndKey: Pkey?,
+  backfillRangeStart: Pkey,
+  backfillRangeEnd: Pkey,
+  scanSize: Long?,
 ): Pkey? {
   // Hibernate doesn't support subqueries in FROM, and we don't want to read in 100k+ records,
   // so we use raw SQL here.
   val pkeyName = backfill.primaryKeyName()
+  val params = mutableListOf<Pkey>()
   var where = when {
-    previousEndKey != null -> "WHERE $pkeyName > ${previousEndKey.utf8()}"
-    else -> "WHERE $pkeyName >= ${backfillRange.start.utf8()}"
+    previousEndKey != null -> {
+      params.add(previousEndKey)
+      "WHERE $pkeyName > ?"
+    }
+    else -> {
+      params.add(backfillRangeStart)
+      "WHERE $pkeyName >= ?"
```

Comment:
wouldn't named parameters be be better? The '?' makes me nervous.

Reply:
I can't find a way to use named params and get the right return type. Have to use a `PreparedStatement` to use the nullSafeGet later on, and it doesn't support naming like createNativeQuery does